### PR TITLE
ISSUE-877: Support Oracle Database

### PR DESCRIPTION
**Pull request by raju-saravanan**
Add Oracle Database support to Streamline.

### CODE REVIEW

File: storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/provider/oracle/factory/OracleExecutor.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    @Override
+    public Long nextId(String namespace) {
+        OracleSequenceIdQuery oracleSequenceIdQuery = new OracleSequenceIdQuery(namespace, connectionBuilder, queryTimeoutSecs, ORACLE_DATA_TYPE_CONTEXT);
+        return oracleSequenceIdQuery.getNextID();
+    }
+
+    @Override
+    public boolean isColumnInNamespace(String namespace, String columnName) throws SQLException {
```
[the remark this answers is not in the document]
Comment (raju-saravanan) [reply storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/provider/oracle/factory/OracleExecutor.java]: Oracle by default uppercases the table's name, so I need to escape the table's name with double quotes, but this syntax won't work for MySQL as the escape quote for MySQL is back tick. So, I have to give a separate implementation for Oracle.

**Comment by HeartSaVioR** [comment]
Just to be clear: The body of PR will become the commit message.

File: conf/streamline-dev.yaml [hunk 2]
```diff
@@ -103,6 +103,20 @@ storageProviderConfiguration:
 #     dataSource.user: "streamline_user"
 #     dataSource.password: "streamline_password"
 
+# --
+# Oracle based storage provider configuration
```
[the remark this answers is not in the document]
Comment (raju-saravanan) [reply conf/streamline-dev.yaml]: I have raised a Jira task for this (https://hortonworks.jira.com/browse/BUG-89400?filter=-1),  please let me know whom should I assign it to.

**Comment by arunmahadevan** [comment]
@raju-saravanan, can you rebase with master and rebuild?. There seems to be some compilation issues in travis when building with latest master.

**Comment by arunmahadevan** [comment]
+1

**Pull request closed without merging**